src/java.base/macosx/native/libjli/java_md_macosx.m

```diff
@@ -722,6 +723,16 @@ static void MacOSXStartup(int argc, char *argv[]) {
     return (void*)(intptr_t)JavaMain(args);
 }
 
+static size_t alignUp(size_t stack_size) {
+    long page_size = sysconf(_SC_PAGESIZE);
```

Comment:
In hotspot we use `getpagesize()`. There is also a guard for a very large stack (within a page of SIZE_MAX) so that rounding up does not produce zero.

Reply:
sounds like that (getpagesize) should work with m1 mac as well, as they have 16k pages. will it ?

Reply:
sysconf is the portable way based on POSIX, we can use getpagesize give this is macOS specific code, which is BSD based.